File: utils/imagenet.py
from pathlib import Path
import pickle

import numpy as np


def unpickle(filepath):
    with filepath.open('rb') as f:
        d = pickle.load(f)
    return d


def shape_batch(
    idx=None, in_template="train_data_batch_{}",
    out_data="train_data_{}.npy",
    out_labels="train_labels_{}.npy"
):
    img_size = 32
    img_size2 = img_size * img_size
    data = unpickle(Path(in_template.format(idx)))
    x = data['data']
    y = data['labels']
    x = np.concatenate((x[:, :img_size2], x[:, img_size2:2*img_size2],
                        x[:, 2*img_size2:]))
    x = x.reshape((x.shape[0], 1, img_size, img_size)).transpose(2, 3, 1, 0)
    y = np.array(y * 3, dtype=np.int)
    np.save(out_data.format(idx), x)
    np.save(out_labels.format(idx), y)


if __name__ == "__main__":
    for idx in range(1, 11):
        shape_batch(idx)

    shape_batch(in_template="val_data", out_data="val_data.npy",
                out_labels="val_labels.npy")

File: utils/imagenet.jl
using PyCall
using JLD

@pyimport numpy as np

function convert_to_jld()
    for idx in 1:10
        x = np.load("train_data_$idx.npy");
        y = np.load("train_labels_$idx.npy");
        save("train_$idx.jld", "data", x, "labels", y)
    end


    x = np.load("val_data.npy");
    y = np.load("val_labels.npy");
    save("val.jld", "data", x, "labels", y)
end

function create_sample_data()
    for idx in 1:10
        data = load("train_$idx.jld")
        x, y = data["data"], data["labels"]
        x = x[:, :, :, 1:100]
        y = y[1:100]
        save("samples/train_$idx.jld", "data", x, "labels", y)
    end

    data = load("val.jld")
    x, y = data["data"], data["labels"]
    x = x[:, :, :, 1:200]
    y = y[1:200]
    save("samples/val.jld", "data", x, "labels", y)
end

create_sample_data()
